src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.TextWriter.cs

```diff
@@ -0,0 +1,496 @@
+
+            public ODataUtf8JsonTextWriter(ODataUtf8JsonWriter jsonWriter)
+            {
+                this.jsonWriter = jsonWriter;
+            }
+
+            public override Encoding Encoding => throw new NotImplementedException();
+
+            /// <summary>
+            /// Flushes any buffered data to the underlying stream synchronously.
+            /// </summary>
+            public override void Flush()
+            {
+                this.jsonWriter.Flush();
+            }
+
+            /// <summary>
+            /// Flushes any buffered data to the underlying stream asynchronously.
+            /// </summary>
+            /// <param name="cancellationToken">A cancellation token to observe while waiting for the flush operation to complete.</param>
+            /// <returns>A task representing the asynchronous flush operation.</returns>
+            public override async Task FlushAsync()
+            {
+                await this.jsonWriter.FlushAsync();
+            }
+
+            /// <summary>
+            /// Disposes the object.
+            /// </summary>
+            /// <param name="disposing">True if called from Dispose; false if called form the finalizer.</param>
+            protected override void Dispose(bool disposing)
+            {
+                this.Flush();
+
+                // If there are unprocessed chars, encode and write them as the final block.
+                ReadOnlySpan<char> charsNotProcessedFromPreviousChunk = this.buffer.AsSpan().Slice(0, numOfCharsNotWrittenFromPreviousChunk);
+
+                if (!charsNotProcessedFromPreviousChunk.IsEmpty)
+                {
+                    if (!this.jsonWriter.IsWritingJson)
+                    {
+                        int firstIndexToEscape = this.jsonWriter.NeedsEscaping(charsNotProcessedFromPreviousChunk);
+                        this.WriteChunk(charsNotProcessedFromPreviousChunk, charsNotProcessedFromPreviousChunk.Length, firstIndexToEscape, true);
+
+                        // set numOfCharsNotWrittenFromPreviousChunk to 0
+                        this.numOfCharsNotWrittenFromPreviousChunk = 0;
+                    }
+                    else 
+                    {
+                        WriteChunkWithoutEscaping(charsNotProcessedFromPreviousChunk, charsNotProcessedFromPreviousChunk.Length, true);
+                    }
+                }
+
+                if (this.buffer != null)
+                {
+                    ArrayPool<char>.Shared.Return(this.buffer);
+                }
+
+                base.Dispose(disposing);
+            }
+
+            /// <summary>
+            /// Asynchronously disposes of the current object and performs cleanup operations.
+            /// </summary>
+            /// <returns>A <see cref="ValueTask"/> representing the asynchronous operation.</returns>
+            public override async ValueTask DisposeAsync()
+            {
+                await this.FlushAsync();

```

Comment:
It think we should only flush after writing content. I don't think it's useful to flush before and after in this case. Also, consider adding `.ConfigureAwait(false)` to async operations.